src/main/scala/mesosphere/marathon/core/storage/store/impl/memory/InMemoryPersistenceStore.scala

```diff
@@ -148,6 +148,26 @@ class InMemoryPersistenceStore(implicit
 
     Future.successful(Done)
   }
+
+  @volatile private[this] var migrationInProgress: Boolean = false
```

Comment:
What do you think about using AtomicBoolean here?

Reply:
Done.